kernel/core/maths/matfuncs.hpp

```diff
@@ -3567,6 +3567,174 @@ static Matrix matrixFilter(const MatFuncData& funcData, const MatFuncErrorInfo&
+
+    std::rotate(matData.begin(), matData.begin() + shift, matData.end());
+
+    Matrix _mResult = Matrix();
+    _mResult.assign(funcData.mat1.rows(), funcData.mat1.cols(), matData);
+
+    return _mResult;
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Function to perform a circular shift
+/// of the elements in a matrix by n columns.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix circShiftCol(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Transpose the matrix (including restructuring the memory) to get a row wise representation of the data
+    Matrix _mResult = Matrix(funcData.mat1);
+    _mResult.transpose();
+    _mResult.extend(_mResult.rows(), _mResult.cols());
+
+    MatFuncData funcDataTransposed = MatFuncData(_mResult, funcData.nVal);
+
+    // Apply the circShiftRow to the transposed matrix
+    _mResult = circShiftRow(funcDataTransposed, errorInfo);
+
+    // Reverse the transposition
+    _mResult.transpose();
+    _mResult.extend(_mResult.rows(), _mResult.cols());
+
+    // TODO: The double transpose is not very efficient. Probably a loop with directly addressing the new elements would be faster.
+
+    return _mResult;
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Wrapper for the two circular shift
+/// functions. Selects the methode and checks
+/// the dimension argument.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix circShift(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Check the dim argument
+    if (funcData.mVal < 0 || funcData.mVal > 1)
+        throw SyntaxError(SyntaxError::INVALID_AXIS, errorInfo.command, errorInfo.position);
+
+    // Call the proper function
+    if (funcData.mVal == 0)
+        return circShiftRow(funcData, errorInfo);
+    else
+        return circShiftCol(funcData, errorInfo);
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Function to perform a shift
+/// of the columns in a matrix by n columns.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix vectShiftCol(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Matrix data is a column-wise vector
+    std::vector<mu::value_type> matData = funcData.mat1.data();
+
+    // Column shift to the right (negative values to the left)
+    for (size_t col = 0; col < funcData.mat1.cols(); col++)
+    {
+        // TODO: Remove the unnecessary int casts
+        int col_old = (int)((((int)col - (int)funcData.nVal) % (int)funcData.mat1.cols()) + (int)funcData.mat1.cols()) % (int)funcData.mat1.cols();

```

Comment:
Same as before, store some of them as  `int`, that makes this line more readable